crates/ra_proc_macro_srv/src/dylib.rs

```diff
@@ -23,40 +23,23 @@ fn is_derive_registrar_symbol(symbol: &str) -> bool {
 fn find_registrar_symbol(file: &Path) -> io::Result<Option<String>> {
     let file = File::open(file)?;
     let buffer = unsafe { Mmap::map(&file)? };
-    let object = Object::parse(&buffer).map_err(invalid_data_err)?;
-
-    let name = match object {
-        Object::Elf(elf) => {
-            let symbols = elf.dynstrtab.to_vec().map_err(invalid_data_err)?;
-            symbols.into_iter().find(|s| is_derive_registrar_symbol(s)).map(&str::to_owned)
-        }
-        Object::PE(pe) => pe
-            .exports
-            .iter()
-            .flat_map(|s| s.name)
-            .find(|s| is_derive_registrar_symbol(s))
-            .map(&str::to_owned),
-        Object::Mach(Mach::Binary(binary)) => {
-            let exports = binary.exports().map_err(invalid_data_err)?;
-            exports
-                .iter()
-                .map(|s| {
-                    // In macos doc:
-                    // https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/dlsym.3.html
-                    // Unlike other dyld API's, the symbol name passed to dlsym() must NOT be
-                    // prepended with an underscore.
-                    if s.name.starts_with("_") {
-                        &s.name[1..]
-                    } else {
-                        &s.name
-                    }
-                })
-                .find(|s| is_derive_registrar_symbol(s))
-                .map(&str::to_owned)
-        }
-        _ => return Ok(None),
-    };
-    return Ok(name);
+    let object = object::File::parse(&*buffer).map_err(invalid_data_err)?;
+    Ok(object
+        .symbols()
```

Comment:
This should probably be

```suggestion
        .dynamic_symbols()
```

Otherwise it won't work if somebody decides to strip the symbols from the proc macro for whatever reason.